extractor/src/main/scala/com/digitalasset/extractor/json/JsonConverters.scala

```diff
@@ -3,92 +3,59 @@
-
-  private val emptyRecord = V.ValueRecord(None, ImmArray.empty).asJson
+  private[this] def sprayToCirce(s: spray.json.JsValue): Json = {
+    import spray.{json => sj}
+    s match {
+      case sj.JsString(v) => Json fromString v
+      case sj.JsNumber(v) => Json fromBigDecimal v
+      case sj.JsBoolean(v) => Json fromBoolean v
+      case sj.JsObject(v) => Json fromFields (v transform ((_, e) => sprayToCirce(e)))
+      case sj.JsArray(v) => Json fromValues (v map sprayToCirce)
+      case sj.JsNull => Json.Null
+    }
+  }
 
-  // TODO it might be much more performant if exploded into separate vals
-  implicit def valueEncoder[T <: LedgerValue]: Encoder[T] = {
-    case r @ V.ValueRecord(_, _) => r.asJson
-    case v @ V.ValueVariant(_, _, _) => v.asJson
-    case V.ValueEnum(_, constructor) => constructor.asJson
-    case V.ValueList(value) => value.asJson
-    case V.ValueOptional(value) => value.asJson
-    case V.ValueMap(value) => value.asJson
-    case V.ValueBool(value) => value.asJson
-    case V.ValueContractId(value) => value.asJson
-    case V.ValueInt64(value) => value.asJson
-    case V.ValueNumeric(value) => LfNumeric.toUnscaledString(value).asJson
-    case V.ValueText(value) => value.asJson
-    case V.ValueTimestamp(value) => value.asJson
-    case V.ValueParty(value) => value.asJson
-    case V.ValueDate(value) => value.asJson
-    case V.ValueUnit => emptyRecord
+  def toJsonString[A: Encoder](a: A): String = {
+    a.asJson.noSpaces
   }
 
-  implicit def frontStackEncoder[A: Encoder]: Encoder[FrontStack[A]] =
-    _.toImmArray.map(_.asJson).toSeq.asJson
+  implicit val recordEncoder: Encoder[OfCid[V.ValueRecord]] = valueEncoder
 
-  implicit val variantEncoder: Encoder[OfCid[V.ValueVariant]] = {
-    case V.ValueVariant(tycon @ _, ctor, value) =>
-      JsonObject(
-        ctor -> value.asJson
-      ).asJson
-  }
+  implicit def valueEncoder[T <: LedgerValue]: Encoder[T] =
+    t => sprayToCirce(LfValueSprayEnc.apiValueToJsValue(t))
 
-  implicit val scalaOptionEncoder: Encoder[Option[LedgerValue]] = _ match {
-    case None =>
-      JsonObject("None" -> emptyRecord).asJson
-    case Some(value) =>
-      JsonObject("Some" -> value.asJson).asJson
-  }
+  implicit val variantEncoder: Encoder[OfCid[V.ValueVariant]] = valueEncoder
 
-  implicit val mapEncoder: Encoder[SortedLookupList[LedgerValue]] = m =>
-    JsonObject(
-      "Map" ->
-        JsonObject
-          .fromIterable(m.toImmArray.map { case (k, v) => k -> v.asJson }.toSeq)
-          .asJson).asJson
+  implicit val mapEncoder: Encoder[SortedLookupList[LedgerValue]] =
+    valueEncoder.contramap(V.ValueMap(_))
 
   implicit val idKeyEncoder: KeyEncoder[Identifier] = id => s"${id.packageId}@${id.name}"
   implicit val idKeyDecoder: KeyDecoder[Identifier] = StringEncodedIdentifier.unapply
 
-  implicit val idEncoder: Encoder[Identifier] = deriveEncoder[Identifier]
-  implicit val idDecoder: Decoder[Identifier] = deriveDecoder[Identifier]
```

Comment:
Interesting macro interaction: if you set `compileTimeOnly` on these, you get the associated error below on multiTableState*coders' `derive*coder` calls, but if you set `private` on these and `-Ywarn-unused`, you get an unused warning. I hypothesize this is because the private implicit, while perfectly visible to a normal inductive implicit resolution, is invisible to [`resolveInstance` as used in the derive*coder calls](https://github.com/circe/circe/blob/v0.10.0/modules/generic/shared/src/main/scala/io/circe/generic/util/macros/DerivationMacros.scala#L271), so then a default implicit takes precedence, illustrating once again the dangers of incoherent typeclass instances.

By contrast, `idEncoder`, `idDecoder`, &c fail the `compileTimeOnly` test, so they are removed in this PR.